src/layout/actions/layout.rs

```diff
@@ -423,8 +423,13 @@ impl LayoutTree {
                 // NOTE Do _NOT_ use set_layout,
                 // the normalization causes the issue described in #344
                 match self.tree[parent_ix] {
-                    Container::Container { ref mut layout , ..} => {
-                        *layout = new_layout
+                    Container::Container { ref mut layout , ref mut borders, ..} => {
+                        *layout = new_layout;
+                        // This needs to be done manually because
+                        // apparently set_layout cannot be used.
```

Comment:
To  work around issues you ran into with this, use this in the match statement instead:

```rust
match {
    ...
    ref mut container @ Container::Container { .. } => {
        container.set_layout(layout)?;
    }
}
````

Reply:
That's not the problem, a few lines up it says in a comment that set_layout cannot be used because it introduces a bug, and down in `fn set_layout` it's done the same way, so I assumed it can be related to the same bug and I did the same.

Reply:
Ah, ok then. Yeah keep it the way it is then, I'll see if I can improve the comment explaining what the bug is if you do that then.